src/reinforcement/mesh.py

```diff
@@ -0,0 +1,224 @@
+                if addx == True:
+                    mymesh.occ.addPoint(x_cord, y_cord, z_cord, tag=ntp + i)
+
+                else:
+                    mymesh.occ.addPoint(y_cord, x_cord, z_cord, tag=ntp + i)
+                reinf_tags.append(ntp + i)
+        return reinf_tags, i
+
+    i_points = 0
+    reinf_tags_x, i_points = reinforcement_points(
+        n_x + 1, x, y, True, where, i_points
+    )  # for reinforcement in x-direction
+
+    # before adding reinforcement in y-direction, it is necessary to know which was the last point that has been added before
+    ltp_Nx = mymesh.occ.getEntities(0)[-1][-1]  # "last tag point"
+    reinf_tags_y, i_points = reinforcement_points(
+        n_y + 1, y, x, False, where, i_points
+    )  # for y-reinforcement
+    reinf_tags = np.hstack((reinf_tags_x, reinf_tags_y))  # save all reinforcement tags
+
+    # add lines which connect the reinforcement points
+    def reinforcement_lines(reinf_tags, ltp_Nx):
+        line_tags = []  # save line tags for physical group
+        ltl = mymesh.occ.getEntities(1)[-1][-1]  # "last tag line"
+        k = 0
+
+        for i, point_tag in enumerate(reinf_tags):
+            if i % 2 != 0:  # otherwise you will connect all points twice
+                k += 1  # to find the correct tags
+                mymesh.occ.addLine(reinf_tags[i - 1], point_tag, tag=ltl + k)
+                line_tags.append(ltl + k)
+        return line_tags
+
+    line_tags = reinforcement_lines(reinf_tags, ltp_Nx)
+
+    # add new points to mesh by synchronizing
+    mymesh.occ.synchronize()
+
+    # add all reinforcement lines to a physical group
+    mymesh.addPhysicalGroup(dim=1, tags=line_tags, tag=1)
+    mymesh.occ.synchronize()
+
+    # now generate the 3D-mesh
+    meshFact = gmsh.model.mesh
+    meshFact.generate(3)
+
+    # Save mesh as msh file
+    gmsh.write(filename)
+    gmsh.finalize
+
+
+# This function takes a msh file as input and separates the physical curve (reinforcement) from the physical volume (concrete) and saves each of them in a separated xdmf file
+def create_xdmf(msh_file):
+    def create_mesh(mesh, cell_type):
+        cells = mesh.get_cells_type(cell_type)
+        cell_data = mesh.get_cell_data("gmsh:physical", cell_type)
+        points = mesh.points
+        out_mesh = meshio.Mesh(
+            points=points,
+            cells={cell_type: cells},
+            cell_data={"name_to_read": [cell_data]},
+        )
+        return out_mesh
+
+    # Read mesh
+    msh = meshio.read(msh_file)
+
+    # Create and save one file containing the whole volume (hexahedra), and one file for the reinforcement lines
+    hexa_mesh = create_mesh(msh, "hexahedron")
+    line_mesh = create_mesh(msh, "line")
+
+    meshio.write("concrete_mesh.xdmf", hexa_mesh)
+    meshio.write("rebar_mesh.xdmf", line_mesh)
+
+    with dfx.io.XDMFFile(MPI.COMM_WORLD, "concrete_mesh.xdmf", "r") as xdmf:
```

Comment:
I feel like creating the xdmf and reading it should be two different functions. Currently I also can't imagine a workflow, where you would ever not need the xdmf in FEniCSx, therefore we maybe should create it directly in `create_concrete_slab`